src/rideables/Stack.hpp

```diff
@@ -0,0 +1,71 @@
+// #ifndef STACK_HPP
+// #define STACK_HPP
+
+#include <stdio.h>
+#include<iostream>
+#include <atomic>
+
+using namespace std;
+
+class Stack
+{
+    private:
+
+    typedef struct StackNode
+    {
+
+        int data;
+        StackNode *next;
+
+    } * StackNodePtr;
+
+    atomic<StackNodePtr> top;
+
+    public:
+    Stack()
+    {
+        top = NULL;
+    }
+    void push(int data){
+
+        StackNodePtr new_node = new StackNode;
+        new_node->data = data;
+        new_node->next = NULL;
+
+        StackNodePtr old_node;
+
+        do {
+            old_node = top;
```

Comment:
When reading from `atomic`, use `load()` instead of directly access it. (so it becomes `old_node = top.load();`)
Direct access won't go wrong on x64, but may be incorrect on other platforms with weaker memory model, since the direct access doesn't come with memory fences for ordering.